src/csharp/Microsoft.Spark/Utils/UdfSerDe.cs

```diff
@@ -197,16 +266,15 @@ private static TypeData SerializeType(Type type)
         /// <returns>The loaded assembly</returns>
         private static Assembly LoadAssembly(string manifestModuleName)
         {
-            var sep = Path.DirectorySeparatorChar;
             try
             {
-                return Assembly.LoadFrom(
-                    $"{Directory.GetCurrentDirectory()}{sep}{manifestModuleName}");
+                return AssemblyLoader(
+                    Path.Combine(Directory.GetCurrentDirectory(), manifestModuleName));
```

Comment:
Would it be a bad idea to check `File.Exists` before the `try-catch`? If there are many calls here, throwing and catching exceptions isn't the best.

Reply:
Sounds like a good idea.  I implemented your suggestion.  Also got rid of the try/catch while deserializing ValueData by utilizing a cache.